server/controllers/wazuh-hosts.js

```diff
@@ -47,21 +50,30 @@ export class WazuhHostsCtrl {
    * @param {Object} hosts
    * @param {Object} registry
    */
-  joinHostRegistry(hosts, registry, removePassword = true) {
+  async joinHostRegistry(hosts, registry, removePassword = true) {
     try {
       if (!Array.isArray(hosts)) {
         throw new Error('Hosts configuration error in wazuh.yml');
       }
 
-      const joined = [];
-      hosts.forEach(h => {
+      return await Promise.all(hosts.map(async h => {
         const id = Object.keys(h)[0];
         const api = Object.assign(h[id], { id: id });
         const host = Object.assign(api, registry[id]);
-        if (removePassword) delete host.password;
-        joined.push(host);
-      });
-      return joined;
+        // Check if api.run_as is false or undefined, then it set to false in cache
+        if(!api.run_as){
+          CacheInMemoryAPIUserAllowRunAs.set(id, api.username, false);
+        };
+        // Add to run_as from API user. Use the cached value or get it doing a request
+        host.allow_run_as = CacheInMemoryAPIUserAllowRunAs.has(id, api.username)
```

Comment:
isn't the same validation into the function checkAPIUserAllowRunAS?

Reply:
I removed the validation, because, the refactorized function is doing the same that here.